packages/blade/src/components/Collapsible/CollapsibleContext.ts

```diff
@@ -9,17 +9,16 @@ type CollapsibleContextState = {
   collapsibleBodyId: string;
 };
 
-// eslint-disable-next-line @typescript-eslint/no-empty-function
-const noop = (): void => {};
+const CollapsibleContext = createContext<CollapsibleContextState | null>(null);
 
-const CollapsibleContext = createContext<CollapsibleContextState>({
-  isExpanded: false,
-  defaultIsExpanded: false,
-  onExpandChange: noop,
-  direction: 'bottom',
-  collapsibleBodyId: '',
-});
-
-const useCollapsible = (): CollapsibleContextState => useContext(CollapsibleContext);
+const useCollapsible = (): CollapsibleContextState => {
+  const collapsibleContext = useContext(CollapsibleContext);
+  if (!collapsibleContext) {
+    throw new Error(
+      `[Blade: CollapsibleContext]: useCollapsible should be only used within CollapsibleContext`,
```

Comment:
since this error will be thrown on subcomponents probably we can tweak the message?

```suggestion
      '[Blade: CollapsibleContext]: You are trying to use Collapsible sub-components without `Collapsible`. `useCollapsible` should be used only within `Collapsible`,
```

Reply:
Updated in https://github.com/razorpay/blade/pull/1296/commits/aba540e2b1bca9ad74314e79ed908a1067a14ffd